game-core/src/main/java/games/strategy/engine/chat/ChatPanel.java

```diff
@@ -106,4 +108,9 @@ public void setShowChatTime(final boolean showTime) {
   public ChatMessagePanel getChatMessagePanel() {
     return chatMessagePanel;
   }
+
+  @Override
+  public Optional<Component> getViewComponent() {
+    return Optional.of(this);
```

Comment:
(1) no casting, type-safe
(2) decouples view and model, we no longer require for the model and view object to be one and the same, we could return an arbitrary object here for the view.